### DISCUSSION TITLE
How are additional task arguments supposed to work together with shell entrypoints?

### DISCUSSION THREAD

**Discussion by sschuberth** [batect/batect#1355] [Q&A] [answered]
So, we're using a Batect task with a login shell as the entry point, very similar to the example [here](https://batect.dev/docs/reference/config/tasks/#command):

```
containers:
  container-1:
    command: "'echo hello && echo world'"
    entrypoint: /bin/sh -c
```

So the `command` requires quoting as described in those docs. However, if I now [pass additional arguments to a task](https://batect.dev/docs/reference/cli/#pass-additional-arguments-to-a-task) via `--` (like `-- more args`), Batect seems to create an internal command that appends the additional arguments to outside of the quotes (like `"'echo hello && echo world'" -- more args`).

That makes the execution fail for obvious reasons. While a work-around is to create a new task that inlines the additional arguments to inside the quotes of the `command`, I wonder whether the current behavior is intended, or whether Batect should append additional task arguments to inside any quotes within a `command`.

**Comment by charleskorn** [comment, marked as answer]
The behaviour you're seeing is the intended behaviour: additional arguments on the command line are appended to the command's arguments, rather than added into the last argument of the command.

Is there a reason why you're using a shell entrypoint like this? 

If the entrypoint is required, then the only solution I'm aware of is to do as you mentioned and create a shell script that is used as the command that handles the additional arguments in the way that you'd like.

**Reply by sschuberth** [reply]
> Is there a reason why you're using a shell entrypoint like this?

Apparently we're depending on some wrapper scripts that are installed in the environment of the user only. See [here](https://github.com/oss-review-toolkit/ort/commit/5b293e7975c3588743cc61f745b22ccf82524068) for some details.
